.differential used twice in intializer list when using multiple adc's

We use two adc's in our project: the built-in adc of the nrf9160 and an external differential adc (ti ads130, custom driver). 
This causes a problem, because the built-in adc enables  `CONFIG_ADC_CONFIGURABLE_INPUTS`, but the external adc has `zephyr,differential` set in the channels. 

This causes `.differential` to appear twice in [`ADC_CHANNEL_CFG_DT`](https://github.com/zephyrproject-rtos/zephyr/blob/main/include/zephyr/drivers/adc.h#L248) of the external adc:

 - once because the channel is set to differential via `zephyr,differential` in the dts.
 - once because `CONFIG_ADC_CONFIGURABLE_INPUTS`  is set in kconfig because nrfx_adc is enabled.

Our current workaround is to force enable `CONFIG_ADC_CONFIGURABLE_INPUTS` in our main Kconfig file, and then mark the channels as differential by setting `zephyr,input-negative` even though this is not actually configurable on this adc.

My suggestion would be to modify this macro as (pseudocode):
```
.differential = (zephyr,differential || zephyr,input_negative configured) 
```
And not have an if around differential anymore, since it has to always be there anyways. 

Alternatively, the `IF_ENABLED(CONFIG_ADC_CONFIGURABLE_INPUTS)` might be converted to an `#if`  statement, so an `#else` might be used, which also insures that the line occurs only once.

I am not experienced enough to judge consequences in zephyr, but if people agree, i wouldn't mind writing this.

the actual error looks like this:
```
In file included from /home/david/dent/zaphod/ncs/zephyr/include/zephyr/sys/util_macro.h:34,
                 from /home/david/dent/zaphod/ncs/zephyr/include/zephyr/irq_multilevel.h:15,
                 from /home/david/dent/zaphod/ncs/zephyr/include/zephyr/devicetree.h:20,
                 from /home/david/dent/zaphod/ncs/zephyr/include/zephyr/device.h:12,
                 from /home/david/dent/zaphod/src/sensors/dtssensors.hpp:5,
                 from /home/david/dent/zaphod/src/sensors/dtssensors.cpp:1:
/home/david/dent/zaphod/ncs/zephyr/include/zephyr/drivers/adc.h:235:29: error: '.differential' designator used multiple times in the same initializer list
  235 |         (.differential    = true,)) \
      |                             ^~~~
/home/david/dent/zaphod/ncs/zephyr/include/zephyr/sys/util_internal.h:72:26: note: in definition of macro '__DEBRACKET'
   72 | #define __DEBRACKET(...) __VA_ARGS__
      |                          ^~~~~~~~~~~
/home/david/dent/zaphod/ncs/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
/home/david/dent/zaphod/ncs/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
/home/david/dent/zaphod/ncs/zephyr/include/zephyr/sys/util_macro.h:180:9: note: in expansion of macro 'Z_COND_CODE_1'
  180 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
/home/david/dent/zaphod/ncs/zephyr/include/zephyr/sys/util_macro.h:224:9: note: in expansion of macro 'COND_CODE_1'
  224 |         COND_CODE_1(_flag, _code, ())
      |         ^~~~~~~~~~~
/home/david/dent/zaphod/ncs/zephyr/include/zephyr/drivers/adc.h:310:9: note: in expansion of macro 'IF_ENABLED'
  310 |         IF_ENABLED(DT_NODE_EXISTS(node_id), \
      |         ^~~~~~~~~~
/home/david/dent/zaphod/ncs/zephyr/include/zephyr/sys/util_internal.h:69:53: note: in expansion of macro '__DEBRACKET'
   69 | #define __GET_ARG2_DEBRACKET(ignore_this, val, ...) __DEBRACKET val
      |                                                     ^~~~~~~~~~~
/home/david/dent/zaphod/ncs/zephyr/include/zephyr/sys/util_internal.h:64:9: note: in expansion of macro '__GET_ARG2_DEBRACKET'
   64 |         __GET_ARG2_DEBRACKET(one_or_two_args _if_code, _else_code)
      |         ^~~~~~~~~~~~~~~~~~~~
/home/david/dent/zaphod/ncs/zephyr/include/zephyr/sys/util_internal.h:59:9: note: in expansion of macro '__COND_CODE'
   59 |         __COND_CODE(_XXXX##_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~
/home/david/dent/zaphod/ncs/zephyr/include/zephyr/sys/util_macro.h:180:9: note: in expansion of macro 'Z_COND_CODE_1'
  180 |         Z_COND_CODE_1(_flag, _if_1_code, _else_code)
      |         ^~~~~~~~~~~~~
/home/david/dent/zaphod/ncs/zephyr/include/zephyr/sys/util_macro.h:224:9: note: in expansion of macro 'COND_CODE_1'
  224 |         COND_CODE_1(_flag, _code, ())
      |         ^~~~~~~~~~~
/home/david/dent/zaphod/ncs/zephyr/include/zephyr/drivers/adc.h:234:1: note: in expansion of macro 'IF_ENABLED'
  234 | IF_ENABLED(DT_PROP(node_id, zephyr_differential), \

```

--- a/include/zephyr/drivers/adc.h
+++ b/include/zephyr/drivers/adc.h
@@ -241,12 +241,13 @@ struct adc_channel_cfg {
 	.reference        = DT_STRING_TOKEN(node_id, zephyr_reference), \
 	.acquisition_time = DT_PROP(node_id, zephyr_acquisition_time), \
 	.channel_id       = DT_REG_ADDR(node_id), \
-IF_ENABLED(CONFIG_ADC_CONFIGURABLE_INPUTS, \
-	(.differential    = DT_NODE_HAS_PROP(node_id, zephyr_input_negative), \
-	 .input_positive  = DT_PROP_OR(node_id, zephyr_input_positive, 0), \
-	 .input_negative  = DT_PROP_OR(node_id, zephyr_input_negative, 0),)) \
-IF_ENABLED(DT_PROP(node_id, zephyr_differential), \
+IF_ENABLED(UTIL_OR(DT_PROP(node_id, zephyr_differential), \
+		   UTIL_AND(CONFIG_ADC_CONFIGURABLE_INPUTS, \
+			    DT_NODE_HAS_PROP(node_id, zephyr_input_negative))), \
 	(.differential    = true,)) \
+IF_ENABLED(CONFIG_ADC_CONFIGURABLE_INPUTS, \
+	(.input_positive  = DT_PROP_OR(node_id, zephyr_input_positive, 0), \
+	 .input_negative  = DT_PROP_OR(node_id, zephyr_input_negative, 0),)) \
 IF_ENABLED(CONFIG_ADC_CONFIGURABLE_EXCITATION_CURRENT_SOURCE_PIN, \
 	(.current_source_pin_set = DT_NODE_HAS_PROP(node_id, zephyr_current_source_pin), \
 	 .current_source_pin = DT_PROP_OR(node_id, zephyr_current_source_pin, {0}),)) \

